File: objc/Bridge/Bridge/Circle.m
#import "Circle.h"

@implementation Circle

-(instancetype)initWith:(int)x y:(int)y radius:(int)r graphicsApi:(id<GraphicsAPI>)api {
    if (self = [super initWithGraphicsAPI:api]) {
        _x = x;
        _y = y;
        _radius = r;
    }
    return self;
}

-(void)draw {
    [self.graphicsApi drawCircle:_x y:_y radius:_radius];
}

@end

File: swift/Bridge/Bridge/Circle.swift
import Foundation

public class Circle : Shape {
    var x: Int = 0
    var y: Int = 0
    var radius: Int = 0
    
    convenience init(_ x: Int, _ y: Int, _ radius: Int, _ graphicsApi: GraphicsAPI) {
        self.init(graphicsApi)
        self.x = x
        self.y = y
        self.radius = radius
    }
    
    override func draw() -> Void {
        self.graphicsApi.drawCircle(self.x, self.y, self.radius)
    }
}
